Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs

```diff
@@ -209,3 +214,15 @@ tfCustomer res =
                 personTags = Nothing
               }
       }
+
+tfCancellationTerms :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => DConfirm.DConfirmResp -> m [Spec.CancellationTerm]
+tfCancellationTerms res = do
+  let vehicleCategory = Utils.mapVariantToVehicle res.vehicleVariant
+  becknConfig <- QBC.findByMerchantIdDomainAndVehicle (Just res.transporter.id) (show Context.MOBILITY) vehicleCategory >>= fromMaybeM (InternalError "Beckn Config not found")
```

Comment:
Keep transformers functions pure. Pass the relevant data to the transformers

Reply:
Fixed this.